src/Microsoft.AspNetCore.Proxy/ProxyAdvancedExtensions.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Proxy
+{
+    static class ProxyAdvancedExtentions
```

Comment:
In my opinion it is a bad practice not to seal classes. Only very small subset of classes is designed to be inherited. Many modern objective-functional languages even prohibit inheritance, but that's different discussion. OK, nvm. I will comply.